### PR TITLE
Implement api calls

### PR DESCRIPTION
**Pull request by ahmedtaa**
# In this branch :
- Refactor your add book and remove book features

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -11,6 +15,23 @@ export const removeBook = (payload) => ({
...
+    Object.keys(data).forEach((book) => {
+      dispatch({
+        type: ADD_BOOK,
+        payload: {
+          item_id: book,
+          ...data[book][0],
+        },
+      });
+    });
+  });
```
Comment (BrianSammit) [inline src/redux/books/books.js:32]: - [x] Kindly make sure that the book you created is stored in the API. Currently, if I created a book it does not store in the API thus the book disappears if I refresh the page. 😄 👍 

Capture

Capture

**Review by BrianSammit — CHANGES_REQUESTED** [review]
# CHANGES REQUIRED 🔴 🛑
Hello @ahmedtaa, great job you are doing so far in this project. 💯

TO HIGHLIGHT:

- Your project is professional 
- Your repo is professional 
- Well structured files 

However, there are a few changes to make to move to the next project 🛠️ . Keep working hard 💪 💻 

## Required Changes ♻️
See my inline comment to know what changes are needed. 👇 👀 

Please make these required changes and send it again for another review 👍 💪
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
# HAPPY CODING!!! 😄 

cat


**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**

--------

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/redux/books/books.js (hunk 1 above)
Comment (ahmedtaa) [reply src/redux/books/books.js:32]: Done

**Comment by omar-labana** [comment]
# **Approved ✔️**

Hi @ahmedtaa,

Congratulations on completing this project! 👏🏼 🎉

You have correctly implemented all of the requirements📃 correctly and without any errors⚠️ or bugs🪲.

Cheers and good luck in your next milestones! 🍀

Feel free to leave any questions or comments in the PR thread(don't forget to tag me 😄) if something is not 100% clear.

---
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**